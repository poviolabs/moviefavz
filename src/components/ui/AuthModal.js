import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

import { Modal, Typography, Button, Divider } from 'antd';

import { Emoji } from '../typography';
import { AuthGraphic } from '../graphics';

const { Title, Paragraph } = Typography;

const StyledGraphicContainer = styled.div`
  display: flex;
  justify-content: center;
  margin: 2rem 0;

  svg {
    width: 75%;
    height: auto;
  }
`;

const AuthModal = ({ visible, onCancel, onConfirm }) => {
  return (
    <Modal
      title="You are not logged in"
      visible={visible}
      onCancel={onCancel}
      onOk={onConfirm}
      footer={null}
    >
      <StyledGraphicContainer>
        <AuthGraphic />
      </StyledGraphicContainer>
      <Title level={3}>
        Hey there, you are not logged in <Emoji emoji="ðŸ‘€" label="eyes" />
      </Title>
      <Paragraph>
        In order to add movies to favorites and view your recent searches, you
        need to authenticate. Click the Login button bellow, to start saving
        your movies.
      </Paragraph>
      <Divider />
      <Button type="primary" block size="large" onClick={onConfirm}>
        Log In
      </Button>
    </Modal>
  );
};

AuthModal.propTypes = {
  visible: PropTypes.bool,
  onCancel: PropTypes.func,
  onConfirm: PropTypes.func,
};

export default AuthModal;
