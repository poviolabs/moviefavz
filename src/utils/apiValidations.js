const validResponse = (status) => status && status >= 200 && status < 300;

export default { validResponse };
