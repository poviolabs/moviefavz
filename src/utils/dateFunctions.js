import dayjs from 'dayjs';

const getUnixDay = (timestamp = Date.now()) => {
  return dayjs(timestamp).startOf('date').valueOf();
};

const fillDaysToMatch = (num, data) => {
  if (!data.length) {
    return [];
  }
  const [lastDate] = data.reverse();

  return Array.from('x'.repeat(num - data.length))
    .map((_, i) => i + 1)
    .reverse()
    .map((i) => {
      return {
        value: 0,
        timestamp: dayjs(lastDate.timestamp).subtract(i, 'day').valueOf(),
      };
    });
};

export default { getUnixDay, fillDaysToMatch };
