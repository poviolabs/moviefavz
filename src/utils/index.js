export { default as arrayFunction } from './arrayFunctions';
export { default as apiValidations } from './apiValidations';
