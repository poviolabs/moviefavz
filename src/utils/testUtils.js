import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import { ThemeProvider } from 'styled-components';
import { theme } from '../styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>{children}</BrowserRouter>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
