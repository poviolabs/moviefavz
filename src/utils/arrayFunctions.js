import dateFunctions from './dateFunctions';

const groupArr = (data, size) => {
  const group = [];
  for (let i = 0, j = 0; i < data.length; i++) {
    if (i >= size && i % size === 0) {
      j++;
    }
    group[j] = group[j] || [];
    group[j].push(data[i]);
  }
  return group;
};

const groupByDay = (data, key = 'timestamp') => {
  const groupedByKey = data.reduce((obj, item) => {
    const timestamp = item[key];
    const day = dateFunctions.getUnixDay(timestamp);

    if (Object.prototype.hasOwnProperty.call(obj, day)) {
      obj[day].value += 1;
    } else {
      obj[day] = {
        value: 1,
        timestamp,
      };
    }
    return obj;
  }, {});
  return Object.values(groupedByKey);
};

export default { groupArr, groupByDay };
