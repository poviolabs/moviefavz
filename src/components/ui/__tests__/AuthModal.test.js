import React from 'react';
import { render } from '../../../utils/testUtils';

import AuthModal from '../AuthModal';

describe('<AuthModal />', () => {
  it('Renders the Component', () => {
    const props = {
      visible: true,
      onCancel: jest.fn(),
      onConfirm: jest.fn(),
    };
    const container = render(<AuthModal {...props} />);
    expect(container).toMatchSnapshot();
  });
});
