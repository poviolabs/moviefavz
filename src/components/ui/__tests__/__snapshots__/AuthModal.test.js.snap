// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthModal /> Renders the Component 1`] = `undefined`;
